SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialHandoverManager.cpp

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "EngineClasses/SpatialHandoverManager.h"
+
+#include "EngineClasses/SpatialPackageMapClient.h"
+#include "SpatialCommonTypes.h"
+#include "SpatialConstants.h"
+#include "SpatialView/EntityDelta.h"
+#include "SpatialView/SpatialOSWorker.h"
+#include "SpatialView/SubView.h"
+#include "SpatialView/ViewDelta.h"
+
+namespace SpatialGDK
+{
+FSpatialHandoverManager::FSpatialHandoverManager(const FSubView& InActorView, const FSubView& InPartitionView)
+	: ActorView(InActorView)
+	, PartitionView(InPartitionView)
+{
+}
+
+void FSpatialHandoverManager::Advance()
+{
+	for (const EntityDelta& Delta : PartitionView.GetViewDelta().EntityDeltas)
+	{
+		switch (Delta.Type)
+		{
+		case EntityDelta::ADD:
+			PartitionsToACK.Add(Delta.EntityId);
+			OwnedPartitions.Add(Delta.EntityId);
+			break;
+		case EntityDelta::REMOVE:
+			OwnedPartitions.Remove(Delta.EntityId);
+			break;
+		case EntityDelta::TEMPORARILY_REMOVED:
+
+			break;
+		default:
+			break;
+		}
+	}
+
+	for (const EntityDelta& Delta : ActorView.GetViewDelta().EntityDeltas)
+	{
+		switch (Delta.Type)
+		{
+		case EntityDelta::UPDATE:
+		{
+			for (const ComponentChange& Change : Delta.ComponentUpdates)
+			{
+				ApplyComponentUpdate(Delta.EntityId, Change.ComponentId, Change.Update);
+			}
+			for (const ComponentChange& Change : Delta.ComponentsRefreshed)
+			{
+				ApplyComponentRefresh(Delta.EntityId, Change.ComponentId, Change.CompleteUpdate.Data);
+			}
+			break;
+		}
+		case EntityDelta::ADD:
+			PopulateDataStore(Delta.EntityId);
+			break;
+		case EntityDelta::REMOVE:
+			DataStore.Remove(Delta.EntityId);
+			break;
+		case EntityDelta::TEMPORARILY_REMOVED:
+			DataStore.Remove(Delta.EntityId);
+			PopulateDataStore(Delta.EntityId);
+			break;
+		default:
+			break;
+		}
+	}
+}
+
+void FSpatialHandoverManager::PopulateDataStore(const Worker_EntityId EntityId)
+{
+	LBComponents2& Components = DataStore.Emplace(EntityId, LBComponents2{});
+	for (const ComponentData& Data : ActorView.GetView()[EntityId].Components)
+	{
+		switch (Data.GetComponentId())
+		{
+		case SpatialConstants::AUTHORITY_INTENTV2_COMPONENT_ID:
+			Components.Intent = AuthorityIntentV2(Data.GetUnderlying());
+			break;
+		case SpatialConstants::AUTHORITY_INTENT_ACK_COMPONENT_ID:
+			Components.IntentACK = AuthorityIntentACK(Data.GetUnderlying());
+			break;
+		default:
+			break;
+		}
+	}
+	HandleChange(EntityId, Components);
+}
+
+void FSpatialHandoverManager::HandleChange(Worker_EntityId EntityId, const LBComponents2& Components)
+{
+	if (Components.Intent.AssignmentCounter != Components.IntentACK.AssignmentCounter)
```

Comment:
Could warn if intent < ack and still process the update. This type of issue occurred in the rpcs recently and might have gone unnoticed without the output.

Reply:
If we add in messages like this they should be errors rather than warnings.

Reply:
Okay, lets put an error in